src/transformers/tokenization_utils_base.py

```diff
@@ -1781,6 +1781,8 @@ def _get_padding_truncation_strategies(
                 padding_strategy = PaddingStrategy.LONGEST  # Default to pad to the longest sequence in the batch
             elif not isinstance(padding, PaddingStrategy):
                 padding_strategy = PaddingStrategy(padding)
+            elif isinstance(padding, PaddingStrategy):
+                padding_strategy = padding
```

Comment:
Ok, let's do the same for the truncation line 1809 if you don't mind to add this:
```
elif isinstance(truncation, TruncationStrategy):
                 truncation_strategy = truncation
```

Reply:
done

Reply:
@thomwolf that change break a unit test, shoud I remove the added lines ?